cylc/uiserver/data_store_mgr.py

```diff
@@ -67,40 +71,98 @@ def __init__(self, workflows_mgr):
         self.executors = {}
         self.delta_queues = {}
 
-    async def sync_workflow(self, w_id, *args, **kwargs):
+    def update_contact(self, w_id, contact_data=None):
+        delta = DELTAS_MAP[WORKFLOW]()
+        delta.time = time.time()
+        flow = delta.updated
+        flow.id = w_id
+        if contact_data:
+            # update with contact file data
+            flow.name = contact_data['name']
+            flow.owner = contact_data['owner']
+            flow.host = contact_data[CFF.HOST]
+            flow.port = int(contact_data[CFF.PORT])
+            # flow.pub_port = int(contact_data[CFF.PUBLISH_PORT])
+            flow.api_version = int(contact_data[CFF.API])
+        else:
+            # wipe pre-existing contact-file data
+            flow.owner, flow.name = w_id.split(ID_DELIM)
+            flow.host = ''
+            flow.port = 0
+            # flow.pub_port = 0
+            flow.api_version = 0
+            flow.status = SuiteStatus.Stopped.value
```

Comment:
This `Stopped` attribute needs to be capitalised to `STOPPED`  (crashes UIS syncs otherwise)..